docs/TESTING.md

```diff
@@ -17,28 +17,64 @@ docker run -it \
            --volume $(pwd):/linux-loader \
            rustvmm/dev:v${container_version}
 cd linux-loader/
-cargo test
+cargo test --all-features
 ```
 
-### bzImage test
+## Test Images
 
-As we don't want to distribute an entire kernel bzImage, the `load_bzImage`
-test is ignored by default. In order to test the bzImage support, one needs to
-locally build a bzImage, copy it to the `src/loader` directory and run
-`cargo test`:
+The kernel images used in the unit tests fall into 3 categories:
+- downloaded from known distro sources using the
+  [download script](../.buildkite/download_resources.sh).
+- fuzzing images; these images were generated by fuzzers such as `libfuzzer` or
+  `afl` starting from an existing image already used by the unit tests. These
+  are named with `fuzz_` so that they are easy to identify.
+- generated images (either by building the Linux kernel, or by using other
+  tools such as [ELFIO](https://github.com/serge1/ELFIO)).
+
+### Building the `PE` image
+
+The PE image used by the linux-loader unit tests is built as follows:
 
 ```bash
-# Assuming your linux-loader and linux-stable are both under ${LINUX_LOADER}:
-git clone git://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git ${LINUX_LOADER}/linux-stable
-cd linux-stable
-make bzImage
-cp linux-stable/arch/x86/boot/bzImage ${LINUX_LOADER}/linux-loader/src/loader/
-cd ${LINUX_LOADER}/linux-loader
-container_version=11
-docker run -it \
-           --security-opt seccomp=unconfined \
-           --volume $(pwd):/linux-loader \
-           rustvmm/dev:v${container_version}
-cd linux-loader/
-cargo test
+git clone git://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git
+git checkout v4.20
+
+# To be able to run the following command, you'll need to install
+# flex and bison.
+make allnoconfig
+make Image
+head -c 4096 arch/arm64/boot/Image > test_image.bin
+```
+
+### Generating `ELF` images with [ELFIO](https://github.com/serge1/ELFIO)
+
+All the ELF files used in `linux-loader` are generated using the `ELFIO` tool.
```

Comment:
I think it's worth a mention that `ELFIO` requires a compiler with builtin C++11 support. For `gcc`/`g++`, that'd be at least `g++-6`. I tried to work around it with my older compiler but it wouldn't budge :)

Reply:
Re-reading this I realize that I inadvertently said something dumb - older compilers (pre v6) **can** build C++11 (and newer) syntax as well, but you need to specify `-std=c++11` explicitly. The `g++` commands below work if you add `-std=c++11`. If you, however, try to build ELFIO itself, I can't figure out how to trick the `automake` magic into appending `-std=c++11` in `CPPFLAGS`, so older compilers can't build it. Nevertheless, turns out you don't need to build ELFIO to get the commands below to work.

Reply:
Yap, building ELFIO was kinda hard (i don't even remember what I had to install), but it shouldn't be needed for building the binaries.